designs/enum-shapes.md

```diff
@@ -0,0 +1,272 @@
+# Enum shapes in Smithy
+
+* **Authors**: Michael Dowling, Jordon Phillips
+* **Created**: 2022-02-14
+* **Last updated**: 2022-02-14
+
+## Abstract
+
+This proposal introduces an `enum` shape and `intEnum` shape to Smithy to
+expand the existing `@enum` trait capabilities.
+
+## Motivation
+
+There are two primary motivations:
+
+1. Smithy does not provide the ability to define an enumerated integer, and
+   many serialization formats serialize enums as integers rather than strings.
+2. Enum values are similar to members, but defined in a less flexible way that
+   requires special casing. Enums in Smithy have enum values that are very
+   similar to members, but cannot be targeted by traits or filtered like other
+   members. For example, an `@enum` trait supports the value, name,
+   documentation, tags, and deprecated properties, of which documentation,
+   tags, and deprecated provide functionality that is exactly equivalent to
+   similarly named traits. In order to filter enums out of a model, model
+   transformations need to write special-cased code to deal with enum traits
+   rather than relying on the existing logic used to remove member shapes.
+
+## Proposal
+
+Two new simple shapes will be introduced in Smithy 2.0:
+
+* `enum`: An enumerated type that is represented using string values
+* `intEnum`: An enumerated type that is represented using integer values.
+
+### Enum shape
+
+The enum shape is used to represent a fixed set of string values. The following
+example defines an enum shape:
+
+```
+enum Suit {
+    DIAMOND
+    CLUB
+    HEART
+    SPADE
+}
+```
+
+Each value listed in the enum is a member that implicitly targets
+`smithy.api#Unit`. The string representation of an enum member defaults to the
+member name. The string representation can be customized by applying the
+`@enumValue` trait.
```

Comment:
In the implementation, you add a default `@enumValue` trait if one isn't present nor is the `@enumDefault` present. Should that baked into the spec? Something about how members with neither of these traits are indistinguishable from members that define an explicit `@enumValue` trait? That way, it won't matter that we default the trait, and people can always rely on using things like selectors to get enum values without needing to worry about what the implementation does (if it defaults or not).